docs(custom-views): add OpenCTI custom views library documentation (#57) (#58)
FiligranHQ/xtm-hub-docs: docs/libraries/opencti-custom-views.md

# OpenCTI Custom Views Library

A comprehensive library of custom views is now available on the XTM Hub,
providing seamless access to curated, role-specific intelligence views.

> Currently, the library focuses on custom views that can be directly deployed to OpenCTI platforms and created by our Filigran Team.

Reviewer: ```suggestion
Currently, the library focuses on custom views that can be directly deployed to OpenCTI products and created by our Filigran Team.
```

@Julienger0  check the new updated taxonomy here: https://docs.hub.filigran.io/latest/libraries/custom-dashboards/

- XTM suite is now called XTM platform (OpenCTI, OpenAEV, XTM one)
- Register is now called Connect
- Platform is now called Product.

Download options are readily available for users who prefer manual import processes,
while sharing capability allow you to generate shareable links that facilitate easy
collaboration with team members and external partners.

The custom views library offers several filters to help you find the view that best suits your needs.
You can search by name and filter custom views by **entity type** and by use case to quickly locate the views relevant to your workflow.

### Manual Import to OpenCTI

Organizations that prefer traditional import methods can
easily download desired custom views from the library and manually integrate them
into their OpenCTI platforms. This process involves downloading the custom view JSON file,
navigating to your OpenCTI platform under `Settings → Customization → Entity type → Custom Views`,
and using the `Import a custom view` functionality to upload the configuration.

### One-Click Deployment

The streamlined deployment process represents the most efficient method for integrating library custom views
into your OpenCTI platform.

Before utilizing this functionality, the following prerequisites must be met:

> - Your OpenCTI platform must be properly registered in the XTM Hub (see [OpenCTI registration documentation](../user/opencti-registration.md)).

Reviewer: ```suggestion
- Your OpenCTI product must be properly registered in the XTM Hub (see [OpenCTI registration documentation](../user/opencti-registration.md)).
```

- Your user account must possess the necessary `Manage customization` capability within OpenCTI.

The deployment process is straightforward: select your desired custom view, click the `Deploy in OpenCTI` button,

> choose your target platform if multiple platforms are registered,

Reviewer: ```suggestion
choose your target product if multiple products are registered,
```

> and wait a few seconds until you are redirected to your OpenCTI platform where the custom view is created.

Reviewer: ```suggestion
and wait a few seconds until you are redirected to your OpenCTI product where the custom view is created.
```

The target entity type of the view is passed automatically, so the view is created against the right entity type without any additional configuration.

![Top right buttons](../assets/images/one-click-deploy.png)

### Sharing and Collaboration

The XTM Hub facilitates seamless collaboration through its comprehensive sharing functionality.
Users can generate universal links for any custom view, enabling cross-organization sharing with partners,
clients, or team members without requiring recipients to maintain XTM Hub accounts.
This approach removes barriers to information sharing while maintaining the integrity and
accessibility of intelligence views across different organizational boundaries.

## Technical Requirements and Best Practices

Successful integration with the XTM Hub requires attention to several technical considerations.
Users deploying custom views must maintain appropriate OpenCTI permissions,
including the `Manage customization` capability.

> Platform registration involves enrolling OpenCTI platforms in the XTM Hub.

Reviewer: ```suggestion
Product connection involves registering OpenCTI product in the XTM Hub.
```
